### ISSUE TITLE
Nested \cond sections are not started

### ISSUE THREAD

**Issue by tomchy** [doxygen/doxygen#8802]
**Description**
Based on my observations, the `\cond` command is not handled if it is present in a different `\cond` section, that is not satisfied. What's more surprising, the corresponding `\endcond` is handled, which leads to an unintended termination of the outer section.

I am still far from full understanding of the parser logic, but I cannot find regex that will call `startCondSection` if the `SkipCond` is activated in `src/pre.l`.

**Expected behavior**
The inner `\cond` section would be either added to the stack (`state->condStack`) or the `\endcond` command will be skipped.

**To Reproduce**
1. Apply the patch from https://github.com/doxygen/doxygen/pull/8800
2. Create a file with nested `\cond` commands
3. Do not satisfy expressions, passed to the `\cond` commands
4. Run doxygen on the file. The new warning should be appear.

**Version**
Found in v1.8.17 (Ubuntu 18.04). Unfortunately my input files are not compatible with 1.9.x. The issue should be fairly simple to check in newer versions.

**Additional context**
If needed, I may try to install and reproduce the issue using other versions of doxygen, but first I would like to confirm that this case should be supported bu the tool.

**Attachments**
Sample project, reproducing the issue: [doxygen_test.zip](https://github.com/doxygen/doxygen/files/7245578/doxygen_test.zip)

**Comment by albert-github** [comment]
You gave a description, but to be sure we are always speaking about the same example:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
> Based on my observations, the \cond command is not handled if it is present in a different \cond section, that is not satisfied. What's more surprising, the corresponding \endcond is handled, which leads to an unintended termination of the outer section.

I found a (paper) note to myself where I see that I wrote that nested `\cond` don't work properly. I think the base of the problem should be fixed as well. 

I think you are referring to a situation like
```
/// \file

/// The initial text
/// \cond
/// The text after first cond
/// \cond
/// The text after second cond
/// \endcond
/// The text after first endcond
/// \endcond
/// The text after second endcond
```
and running dunning `doxygen -d preprocessor` we get:
00001 /// \file
00002
00003 /// The initial text
00004 ///
00005
00006
00007
00008
00009 /// The text after first endcond
00010 ///
00011 /// The text after second endcond
00012
```
whilst actually line 9 should be empty.

- Is this the situation you are referring to?

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/7244548/example.tar.gz)

**Comment by tomchy** [comment]
Sample project attached, as requested. Additionally - confirmed to fail on master branch.

**Comment by tomchy** [comment]
Yes, that's exactly the issue!
Thank you for such immediate response :slightly_smiling_face:

**Comment by albert-github** [comment]
Slightly modified example so `\cond` and `\endcond` arn;t in the same comment block: [example1.tar.gz](https://github.com/doxygen/doxygen/files/7244860/example1.tar.gz)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8804.
(As the fix from #8800 was also necessary for the warning I included it as well).

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
